### PR TITLE
Add Qdrant health checks

### PR DESCRIPTION
**Pull request by Alirexaa**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Please reference the issue this PR will close: https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/issues/2186

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

File: src/HealthChecks.Qdrant/HealthChecks.Qdrant.csproj [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    <TargetFrameworks>$(DefaultLibraryTargetFrameworks)</TargetFrameworks>
+    <PackageTags>$(PackageTags);Qdrant</PackageTags>
+    <Description>HealthChecks.Qdrant is the health check package for Qdrant.</Description>
+    <VersionPrefix>$(HealthCheckQdrant)</VersionPrefix>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Qdrant.Client" Version="1.12.0" />
+    <PackageReference Include="Microsoft.Extensions.Diagnostics.HealthChecks" Version="8.0.0" />
+    <!--<Compile Include="../HealthCheckResultTask.cs" />-->
```
Comment (eerhardt) [inline src/HealthChecks.Qdrant/HealthChecks.Qdrant.csproj:13]: Can this be removed?

File: src/HealthChecks.Qdrant/QdrantHealthCheck.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+/// </summary>
+public class QdrantHealthCheck : IHealthCheck
+{
+    private readonly QdrantClient _client;
+
+    public QdrantHealthCheck(QdrantClient? client)
+    {
+        _client = Guard.ThrowIfNull(client);
+    }
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
```
Comment (eerhardt) [inline src/HealthChecks.Qdrant/QdrantHealthCheck.cs:17]:
```suggestion

    /// <inheritdoc />
    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
```

File: src/HealthChecks.Qdrant/QdrantHealthCheck.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+namespace HealthChecks.Qdrant;
+
+/// <summary>
+/// A health check for Qdrant services.
+/// </summary>
+public class QdrantHealthCheck : IHealthCheck
+{
+    private readonly QdrantClient _client;
+
+    public QdrantHealthCheck(QdrantClient? client)
```
Comment (eerhardt) [inline src/HealthChecks.Qdrant/QdrantHealthCheck.cs:13]:
```suggestion
    public QdrantHealthCheck(QdrantClient client)
```

This shouldn't be nullable if we are going to throw when it is null.

File: src/HealthChecks.Qdrant/QdrantHealthCheck.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        _client = Guard.ThrowIfNull(client);
+    }
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var response = await _client.HealthAsync(cancellationToken).ConfigureAwait(false);
+            return !(response is null || response.Title is null)
```
Comment (eerhardt) [inline src/HealthChecks.Qdrant/QdrantHealthCheck.cs:22]:
```suggestion
            return response?.Title is not null
```

File: test/HealthChecks.Qdrant.Tests/Functional/QdrantHealthCheckTests.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                {
+                    Predicate = r => r.Tags.Contains("qdrant")
+                });
+            });
+
+        using var server = new TestServer(webHostBuilder);
+
+        using var response = await server.CreateRequest("/health").GetAsync();
+
+        response.StatusCode.ShouldBe(HttpStatusCode.OK);
```
Comment (eerhardt) [inline test/HealthChecks.Qdrant.Tests/Functional/QdrantHealthCheckTests.cs:31]: For my learning, how does this return OK when talking to the localhost? There isn't a Qdrant instance running, is there?

Comment (Alirexaa) [reply test/HealthChecks.Qdrant.Tests/Functional/QdrantHealthCheckTests.cs:31]: Qdrant instance running by ci action
see: 
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/2310/files#diff-5d6b2fd45c59f6628e08c23dcf560e4b79d0a6484e47de1a258a58d063d2f21aR31-R36

In most cases, we use this approach to run services in this repo.

Comment (eerhardt) [reply test/HealthChecks.Qdrant.Tests/Functional/QdrantHealthCheckTests.cs:31]: But if I just run the test from Test Explorer, it fails, right?

**Review by eerhardt — APPROVED** [review]
This looks great to me. Thanks for pushing this through.

@adamsitnik @unaizorrilla - any thoughts?

File: test/HealthChecks.Qdrant.Tests/Functional/QdrantHealthCheckTests.cs (hunk 5 above)
Comment (Alirexaa) [reply test/HealthChecks.Qdrant.Tests/Functional/QdrantHealthCheckTests.cs:31]: I think so.

File: test/HealthChecks.Qdrant.Tests/DependencyInjection/RegistrationTests.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        var options = serviceProvider.GetRequiredService<IOptions<HealthCheckServiceOptions>>();
+
+        var registration = options.Value.Registrations.First();
+        var check = registration.Factory(serviceProvider);
+
+        registration.Name.ShouldBe("qdrant");
+        check.ShouldBeOfType<QdrantHealthCheck>();
+    }
+
+    private QdrantClient _clientFactory(IServiceProvider _)
```
Comment (adamsitnik) [inline test/HealthChecks.Qdrant.Tests/DependencyInjection/RegistrationTests.cs:62]: I see what you are trying to achieve, but I believe that using an underscore for the name of a private method is a bad idea, as most of the C# devs would expect it to be a field.

File: src/HealthChecks.Qdrant/QdrantHealthCheck.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        _client = Guard.ThrowIfNull(client);
+    }
+
+    /// <inheritdoc />
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            await _client.HealthAsync(cancellationToken).ConfigureAwait(false);
+
```
Comment (adamsitnik) [inline src/HealthChecks.Qdrant/QdrantHealthCheck.cs:24]:
```suggestion

```

File: src/HealthChecks.Qdrant/QdrantHealthCheck.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    /// <inheritdoc />
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            await _client.HealthAsync(cancellationToken).ConfigureAwait(false);
+            return HealthCheckResult.Healthy();
+        }
+        catch (RpcException ex)
```
Comment (adamsitnik) [inline src/HealthChecks.Qdrant/QdrantHealthCheck.cs:26]:
```suggestion
        catch (Exception ex)
```

**Pull request merged**